src/components/PopupWithForm/PopupWithForm.js

```diff
@@ -1,103 +1,27 @@
-import { useState } from "react";
 import Popup from "../Popup/Popup";
+import SignInForm from "../SignInForm/SignInForm";
+import SignUpForm from "../SignUpForm/SignUpForm";
 
 import "./PopupWithForm.css";
 
-function PopupWithForm({ isOpen, close }) {
-  const [isNewUser, setIsNewUser] = useState(false);
-  function switchForm() {
-    setIsNewUser(!isNewUser);
-  }
-
+function PopupWithForm({
+  isOpen,
+  close,
+  openMessage,
+  isRegistered,
+  switchForm,
+}) {
   return (
     <Popup isOpen={isOpen} close={close}>
       <form className="user-form" validate="true">
-        {isNewUser ? (
-          <>
-            <h3 className="user-form__title">Sign up</h3>
-            <label className="user-form__label">
-              Email
-              <input
-                className="user-form__input"
-                type="email"
-                name="email"
-                placeholder="Enter email"
-                minLength="6"
-                maxLength="30"
-              ></input>
-              <span className="user-form__error"></span>
-            </label>
-            <label className="user-form__label">
-              Password
-              <input
-                className="user-form__input"
-                type="password"
-                name="password"
-                placeholder="Enter password"
-                minLength="6"
-                maxLength="30"
-              ></input>
-              <span className="user-form__error"></span>
-            </label>
-            <label className="user-form__label">
-              Username
-              <input
-                className="user-form__input"
-                type="text"
-                name="username"
-                placeholder="Enter your username"
-                minLength="2"
-                maxLength="30"
-              ></input>
-              <span className="user-form__error"></span>
-            </label>
-            <button className="user-form__button" type="submit" disabled>
-              Sign up
-            </button>
-            <p className="user-form__text">
-              or{" "}
-              <span className="user-form__link" onClick={switchForm}>
-                Sign in
-              </span>
-            </p>
-          </>
+        {!isRegistered ? (
```

Comment:
`PopupWithForm` should receive the whole markup as `children` rather that checking ` {!isRegistered ? ( ` inside it. You need to make it universal.